### DISCUSSION TITLE
Is GumTree suitable for software plagiarism detection?

### DISCUSSION THREAD

**Discussion by namasikanam** [GumTreeDiff/gumtree#291] [Ideas]
I am curious about the effectivenss of GumTree for the task of software plagiarism detection. It seems a natural idea to use the length of edit scripts or Jaccard similarity of mappings between two ASTs, so as to compute a similarity score between two source code files.

Update (2022-09-17): I will try to spare time to find a way to thoroughly and fairly evaluate it, if anyone is also interested to this, please contact [me](https://t.me/namasikanam)!

**Comment by jrfaller** [comment]
Hi!

I think it could be possible! Maybe it would require customizing the weight of some actions (for instance low weight for moving code around, low weight for renaming, high weight for adding or deleting code). It would be a fun experiment to assess how it compares to some baseline.

Cheers!

**Reply by namasikanam** [reply]
Thanks for reply and encouragement! I am willing to try to spare time to evaluate it.

I've found some baselines such as [MOSS](https://theory.stanford.edu/~aiken/moss/) and [JPleg](https://github.com/jplag/JPlag), but finding a **fair** benchmark seems not easy. Maybe [this OOPSLA'20 paper](https://dl.acm.org/doi/abs/10.1145/3428206) could give a fair measure, but it is not open-source :(

**Reply by jrfaller** [reply]
Hi!

The OOPSLA'20 paper is a good find! It's written in the text the the tool is not disclosed to avoid cheating IIUIC, but available on request for researchers so I guess you can ask for it. The authors might also have a dataset of student work?
